### DISCUSSION TITLE
Having different route group root layouts

### DISCUSSION THREAD

**Discussion by ioannis-tsirovasilis** [serwist/serwist#31] [Q&A]
Hello,

In NextJS 14, I have split my application into two route groups (loggedIn), (notLoggedIn) as those pages must have a different root layout. (loggedIn) contains a side bar and a nav bar. According to the docs, I have removed the app/layout.tsx and added <html> and <body> tags to the root layouts of both route groups. 

How PWA works with that? Should I define metadata in each layout? Currently, I'm getting hydration errors when moving from a page that uses the (not logged in) layout to one that uses the (loggedIn) layout.

Thank you in advance!

**Comment by DuCanhGH** [comment]
@ioannis-tsirovasilis I am really sorry, but due to the fact that GitHub doesn't notify me of these discussions, it can get quite hard to keep track of things, leading me to forgetting yours.

> In NextJS 14, I have split my application into two route groups (loggedIn), (notLoggedIn) as those pages must have a different root layout. (loggedIn) contains a side bar and a nav bar. According to the docs, I have removed the app/layout.tsx and added and tags to the root layouts of both route groups.

Wondering why you need to do that. Serwist's docs also has two different layouts, but the PWA-related stuff still gets to stay in one shared root layout that doesn't have any UI code.

> How PWA works with that? Should I define metadata in each layout? Currently, I'm getting hydration errors when moving from a page that uses the (not logged in) layout to one that uses the (loggedIn) 

I don't think this issue is related to your app being a PWA. Most likely, there is actually an issue with your code that may lead to a hydration error. You may disable Serwist temporarily to confirm this.

**Reply by ioannis-tsirovasilis** [reply]
@DuCanhGH 
No Worries!

> I am really sorry, but due to the
>  fact that GitHub doesn't notify me of these discussions, it can get quite hard to keep track of things, leading me to forgetting yours.

You are right, it wasn't a problem with PWA.  It was my mistake eventually. 

Thank you for replying!
